pkg/function/backup_data_stats.go

```diff
@@ -71,33 +71,46 @@ func backupDataStats(ctx context.Context, cli kubernetes.Interface, tp param.Tem
 		PodOverride:  podOverride,
 	}
 	pr := kube.NewPodRunner(cli, options)
-	podFunc := backupDataStatsPodFunc(cli, tp, namespace, encryptionKey, backupArtifactPrefix, backupID, mode)
-	return pr.Run(ctx, podFunc)
+	podFunc := backupDataStatsPodFunc(tp, encryptionKey, backupArtifactPrefix, backupID, mode)
+	return pr.RunEx(ctx, podFunc)
 }
 
-func backupDataStatsPodFunc(cli kubernetes.Interface, tp param.TemplateParams, namespace, encryptionKey, backupArtifactPrefix, backupID, mode string) func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
-	return func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
+func backupDataStatsPodFunc(tp param.TemplateParams, encryptionKey, backupArtifactPrefix, backupID, mode string) func(ctx context.Context, pc kube.PodController) (map[string]interface{}, error) {
```

Comment:
```suggestion
func backupDataStatsPodFunc(
	tp param.TemplateParams,
	encryptionKey,
	backupArtifactPrefix,
	backupID,
	mode string,
) func(ctx context.Context, pc kube.PodController) (map[string]interface{}, error) {
```
Formatting change improve readability